docusaurus/docs/publish-a-plugin/build-automation.md

```diff
@@ -182,6 +183,65 @@ If you encounter errors in the plugin validator or your plugin submission like t
 
 Follow the steps above to enable provenance attestation in your GitHub Actions workflow.
 
+## Automatically Generate Changelogs
+
+Maintaining a detailed changelog is essential for communicating updates to your users and is displayed prominently in the Grafana plugin details page. To simplify this process, our plugin build workflow now supports automatic changelog generation.
+
+### Using the GitHub Actions Workflow for Changelog Generation
+
+The build-plugin GitHub Action can automatically generate and maintain your plugin's changelog using the[github-changelog-generator](https://github.com/github-changelog-generator/github-changelog-generator) tool. This feature:
+
+1. Creates a comprehensive CHANGELOG.md file organized by release
+1. Groups changes by type (features, bug fixes, etc.)
+1. Includes links to pull requests and issues
+1. Acknowledges contributors
+1. Commits the updated changelog to your repository
+
+To enable automatic changelog generation in your workflow, add the `use_changelog_generator: true` parameter to your build-plugin action:
+
+```yaml
+- uses: grafana/plugin-actions/build-plugin@main
+  with:
+    policy_token: ${{ secrets.GRAFANA_ACCESS_POLICY_TOKEN }}
+    attestation: true
+    use_changelog_generator: true # Enable automatic changelog generation
+```
+
+### Requirements
```

Comment:
We should add information here about protected branches

Reply:
added at the 221 line